src/ray/gcs/redis_context.cc

```diff
@@ -133,7 +135,71 @@ RedisContext::~RedisContext() {
   }
 }
 
-Status RedisContext::Connect(const std::string &address, int port) {
+static void GetRedisShards(redisContext *context, std::vector<std::string> *addresses,
```

Comment:
There's some overlap with #2281. @heyucongtom can you take a look?

Reply:
We are both doing a translation (from old ray get_redis_shards to new) here. I think the major difference is that 
+ #2298 add an use_sharding bool to indicate if we are using sharding.
- #2281 modifies table.cc so that when tables do the work, they randomly select a redis shard by the ID.
- #2281 attaches all contexts to event loop, while only connecting the main context (I guess it may be wrong: since tests are giving timeouts.)
- #2281 defaultly using all shards.

Reply:
The same thing is: we are writing the exactly same thing in getRedisShards function

Reply:
Sounds right. This PR is actually really doing sharding. It's just creating a single extra shard (as opposed to an arbitrary number of additional shards).